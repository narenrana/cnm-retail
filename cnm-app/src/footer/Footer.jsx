import React from "react";
import PropTypes from "prop-types";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import { Grid } from "@material-ui/core";
import MenuItem from "@material-ui/core/MenuItem";
import MenuList from "@material-ui/core/MenuList";
import Copyright from "./Copyright";
import { footerMenu, appStoreLinks, footerSocialMediaLinks } from "./data";
import useStyles from "./style";
import Subscribe from "./Subscribe";

export default function Footer(props) {
  const classes = useStyles();

  return (
    <footer className={classes.footer}>
      <Container maxWidth="lg">
        <Subscribe />
        <Grid container spacing={2}>
          <Grid item xs={12} md={10} xl={10}>
            <Grid container spacing={0}>
              {footerMenu.map((footer, index) => (
                <Grid
                  item
                  xs={12}
                  md={2}
                  xl={2}
                  key={"foot-menu-link-" + footer.title + index}
                >
                  <Typography variant="subtitle1">{footer.title}</Typography>
                  <MenuList>
                    {footer.items.map((menu) => (
                      <MenuItem
                        key={"foot-items-link-" + menu.label + "-" + index}
                      >
                        {menu.label}
                      </MenuItem>
                    ))}
                  </MenuList>
                </Grid>
              ))}
            </Grid>
          </Grid>
          <Grid item xs={12} md={2} xl={2}>
            <MenuList>
              {appStoreLinks.map((menu, index) => (
                <MenuItem key={"app-store-link-" + index}>
                  <img alt={menu.image} src={menu.image} />
                </MenuItem>
              ))}
            </MenuList>
            <Grid container className={classes.socialLinks}>
              {footerSocialMediaLinks.map((links, index) => (
                <Grid item xs={3} key={"social-link-" + index}></Grid>
              ))}
            </Grid>
          </Grid>
        </Grid>

        <Copyright />
      </Container>
    </footer>
  );
}

Footer.propTypes = {
  description: PropTypes.string,
  title: PropTypes.string,
};
