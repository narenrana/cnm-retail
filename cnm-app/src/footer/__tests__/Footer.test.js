/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import Footer from "../Footer";

describe("Footer test", () => {
  it("Footer should match snapshot", () => {
    const component = renderer.create(
      <Footer description={undefined} title={undefined} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
