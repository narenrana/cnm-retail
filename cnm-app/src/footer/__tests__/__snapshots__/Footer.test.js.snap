// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer test Footer should match snapshot 1`] = `
<footer
  className="makeStyles-footer-1"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div>
      <img
        alt="Banner"
        className="makeStyles-footerBanner-2"
        src="/static/images/banner.png"
      />
      <div>
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-xl-12"
          >
            <p
              className="MuiTypography-root makeStyles-subscribeTypehead-7 MuiTypography-body1"
            >
              Subscribe to our news latter and receive exclusive offer every week
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-xl-8"
          >
            <div
              className="MuiInputBase-root makeStyles-subscribeInput-8"
              onClick={[Function]}
            >
              <input
                className="MuiInputBase-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Enter your email"
                type="text"
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-xl-4"
          >
            <button
              aria-label="subscribe"
              className="MuiButtonBase-root MuiFab-root makeStyles-subscribeButton-9 MuiFab-extended MuiFab-sizeSmall MuiFab-primary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiFab-label"
              >
                <div
                  className="makeStyles-subscribeButtonText-10"
                >
                  Subscribe
                </div>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10 MuiGrid-grid-xl-10"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiGrid-grid-xl-2"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle1"
            >
              Sitename
            </h6>
            <ul
              className="MuiList-root MuiList-padding"
              onKeyDown={[Function]}
              role="menu"
              tabIndex={-1}
            >
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={0}
              >
                Title One
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Title Two
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Title Three
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Title Four
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Title Five
              </li>
            </ul>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiGrid-grid-xl-2"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle1"
            >
              Information
            </h6>
            <ul
              className="MuiList-root MuiList-padding"
              onKeyDown={[Function]}
              role="menu"
              tabIndex={-1}
            >
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={0}
              >
                Category One
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Category Two
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Category Three
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Category Four
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Category Five
              </li>
            </ul>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiGrid-grid-xl-2"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle1"
            >
              Category
            </h6>
            <ul
              className="MuiList-root MuiList-padding"
              onKeyDown={[Function]}
              role="menu"
              tabIndex={-1}
            >
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={0}
              >
                Category One
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Category Two
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Category Three
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Category Four
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Category Five
              </li>
            </ul>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiGrid-grid-xl-2"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle1"
            >
              Category
            </h6>
            <ul
              className="MuiList-root MuiList-padding"
              onKeyDown={[Function]}
              role="menu"
              tabIndex={-1}
            >
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={0}
              >
                help@sitename.com
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Hotline: +78 9089 8000
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiGrid-grid-xl-2"
      >
        <ul
          className="MuiList-root MuiList-padding"
          onKeyDown={[Function]}
          role="menu"
          tabIndex={-1}
        >
          <li
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="menuitem"
            tabIndex={0}
          >
            <img
              alt="/static/images/app-store-apple.png"
              src="/static/images/app-store-apple.png"
            />
          </li>
          <li
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="menuitem"
            tabIndex={-1}
          >
            <img
              alt="/static/images/google-play-badge.png"
              src="/static/images/google-play-badge.png"
            />
          </li>
        </ul>
        <div
          className="MuiGrid-root makeStyles-socialLinks-3 MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <img
              alt="/static/images/002-facebook-logo.png"
              src="/static/images/002-facebook-logo.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <img
              alt="/static/images/004-linkedin-logo.png"
              src="/static/images/004-linkedin-logo.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <img
              alt="/static/images/005-youtube.png"
              src="/static/images/005-youtube.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <img
              alt="/static/images/001-twitter-logo-silhouette.png"
              src="/static/images/001-twitter-logo-silhouette.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <img
              alt="/static/images/instagram.png"
              src="/static/images/instagram.png"
            />
          </div>
        </div>
      </div>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright © 
       
      2020
      ALL RIGHT RESERVED
    </p>
  </div>
</footer>
`;
